dbt/parser.py

```diff
@@ -272,14 +273,27 @@ def parse_sql_nodes(nodes, root_project, projects, tags=None, macros=None):
                              package_name,
                              node.get('name'))
 
-        # TODO if this is set, raise a compiler error
-        to_return[node_path] = parse_node(node,
-                                          node_path,
-                                          root_project,
-                                          projects.get(package_name),
-                                          projects,
-                                          tags=tags,
-                                          macros=macros)
+        node_parsed = parse_node(node,
+                                 node_path,
+                                 root_project,
+                                 projects.get(package_name),
+                                 projects,
+                                 tags=tags,
+                                 macros=macros)
+
+        # Ignore disabled nodes
+        if not node_parsed['config']['enabled']:
+            continue
+
+        # Check for duplicate model names
+        existing_node = to_return.get(node_path)
+        if existing_node is not None:
+            raise dbt.exceptions.CompilationException(
+                'Found models with the same name:\n- %s\n- %s' % (
+                    existing_node.get('original_file_path'),
+                    node.get('original_file_path')))
```

Comment:
do we need this check in addition to the one in `ModelLoader`? if not let's just rip this out to DRY this up

Reply:
I think we still need this check because `to_return` is a dictionary and keys for duplicate models are non-unique. Without this check, only one of the models would be left in the dictionary after this stage and user would not be notified about the existing duplicate.